### DISCUSSION TITLE
voy2isis can't decompress file, even though I already decompressed it using vdcomp

### DISCUSSION THREAD

**Discussion by jescully** [DOI-USGS/ISIS3#5493] [Q&A]
Hello, I am trying to process Voyager images of Umbriel, and I'm having an issue with voy2isis and vdcomp. I downloaded images from the PDS Imaging Node, in the format of c*name*.imq. Then I ran voy2isis as follows:

voy2isis from=input/2473029.imq to=c2473029.cub

But I got the following error:

can't open input file: input/2473029.imq
**I/O ERROR** Unable to decompress input file [2473029.imq].

So then I tried using vdcomp, with each of the 4 different output formats in turn. When I tried these, supposedly decompressed, images through voy2isis I still got the same error about ISIS not being able to decompress the input file. So vdcomp didn't actually decompress it? 

I'd really appreciate it if someone could help! Not sure what I'm missing here.

**Comment by amystamile-usgs** [comment]
The voy2isis program will try to run vdcomp if the inputted file has an .imq file extension. If it fails, the program throws the "Unable to decompress input file" error. If you could provide me the link to where you downloaded the imq file and the version of ISIS you are using, I will try to replicate this on my end.

**Reply by amystamile-usgs** [reply]
Also if you want to try just manually decompressing, run `vdcomp` with option 1 and set the output file as `2473029.img` instead of `2473029.imq`. voy2isis should avoid trying to decompress the file because it is now an .img file extension. So you should not see the same error.

**Comment by jescully** [comment]
Hello, I downloaded the data from here: https://planetarydata.jpl.nasa.gov/img/data/vg2-u-iss-2-edr-v1.0/vg_0002/umbriel/

I'm not sure what version I'm running, but I just downloaded and installed ISIS using anaconda yesterday, so I'm probably using the latest version. (I used to use ISIS with rsync, and just started with this new conda version yesterday.)

I tried changing the output name to .img as you suggested above, and that fixed the original error - hooray! But now I have a new one: 

**ERROR** Error in PVL file on line [12].
**ERROR** Error when reading a pvl: Cannot have ['/*'] inside a multi-line comment.

I went into the .img file and deleted all instances of /*, but then got another new error:

**ERROR** Error in PVL file on line [2].
**ERROR** Unable to read PVL keyword [FILE FORMAT AND LENGTH].
**ERROR** Expected an assignment [=] when reading PVL, but found [F].

Thank you for your help!

**Reply by amystamile-usgs** [reply]
That makes sense why you are seeing these errors. Looks like if you pass in a .imq image to `voy2isis`, it also does a find and replace, replacing "/*" with "# ". 

I just pulled the latest release of isis from anaconda and ran `voy2isis from=c2646640.imq to=~/testData/voy.cub` with no decompression errors. Might be worth rechecking your anaconda build. See instructions here https://astrogeology.usgs.gov/docs/how-to-guides/environment-setup-and-maintenance/installing-isis-via-anaconda/.

Manually going in and replacing `/*` with `# ` is also an option but not an ideal one. Especially if you need to process multiple voyager images. I tried this myself and was able to get voy2isis to run.

**Reply by jescully** [reply]
Thank you! I tried again this morning and got a different error. I didn't run vdcomp on a file first, and just went straight to voy2isis with a .imq file, as follows:

voy2isis from=c2684004.imq to=c2684004.cub
**ERROR** No existing files found with a numerical version matching [naif????.tls] in [/opt/anaconda3/envs/isis/data/base/kernels/lsk].

So I'm getting closer to this working! Do I need to download new kernels?

**Reply by jescully** [reply]
Also, I think I checked the anaconda build, and got the response: # All requested packages already installed.

And I'm running conda 24.4.0. I tried updating it and got the response: # All requested packages already installed.

**Reply by amystamile-usgs** [reply]
Sounds like you got past the first error so that's good! Looks like you need to download the isis data. Here are the instructions for installing: https://github.com/DOI-USGS/ISIS3?tab=readme-ov-file#full-isis-data-download. I would recommend just downloading the base data using `downloadIsisData base $ISISDATA`.

**Reply by jescully** [reply]
Yes, the first error just magically disappeared when I ran it again today.

And I just realized based on your last post that I didn't quite install all the base data on Friday - I just missed some of it when my connection timed out. But I downloaded the rest of it today and now it seems to be working perfectly with none of the above errors. Thank you!
